src/fuzzfetch/fetch.py

```diff
@@ -349,9 +358,51 @@ def __init__(self, target, branch, build, flags, platform=None):
             self._task = build
 
         else:
-            self._task = BuildTask(build, branch, self._flags, self._platform)
-
             now = datetime.now(timezone('UTC'))
+            try:
+                self._task = BuildTask(build, branch, self._flags, self._platform)
+            except FetcherException:
+                if not nearest:
+                    raise
+                else:
+                    asc = nearest == Fetcher.BUILD_ORDER_ASC
+                    start = None
+                    if build == 'latest':
+                        start = now
+                    elif re.match(r'^[0-9[a-f]{12,40}$', build):
+                        start = _get_rev_date(build)
+
+                    elif re.match(r'^[0-9]{4}-[0-9]{2}-[0-9]{2}$', build):
+                        date = datetime.strptime(build, '%Y-%m-%d')
+                        localized = timezone('UTC').localize(date)
+                        start = localized + timedelta(days=1) if asc else localized - timedelta(days=1)
+                    else:
+                        # If no match, assume it's a TaskCluster namespace
+                        if re.match(r'.*(nightly|pushdate).*[0-9]{4}\.[0-9]{2}\.[0-9]{2}', build):
+                            match = re.search(r'[0-9]{4}\.[0-9]{2}\.[0-9]{2}', build)
+                            start = datetime.strptime(match.group(0), '%Y.%m.%d')
+                        elif re.match(r'.*revsion.*[0-9[a-f]{12,40}', build):
+                            match = re.search(r'[0-9[a-f]{12,40}', build)
+                            start = _get_rev_date(match.group(0))
+
+                    # If start date is outside the range of the newest/oldest available build, adjust it
+                    if asc and start < now - timedelta(days=364):
+                        start = now - timedelta(days=364)
+                    elif not asc and start > now:
+                        start = now
+
+                    end = now if asc else now - timedelta(days=364)
+
+                    while start < end if asc else start > end:
```

Comment:
Question: would this be clearer like:

    inc = 1 if asc else -1
    for days in range(0, (end-start).days, inc):
        build_date = start + timedelta(days=days)
        ...

?
I'm not positive the end condition is correct.